staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go

```diff
@@ -0,0 +1,82 @@
+package create
+
+import (
+	"github.com/spf13/cobra"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+	generateversioned "k8s.io/kubectl/pkg/generate/versioned"
+	"k8s.io/kubectl/pkg/generate"
+)
+
+// IngressOpts is returned by NewCmdCreateIngress
+type IngressOpts struct {
+	CreateSubcommandOptions *CreateSubcommandOptions
+}
+
+// NewCmdCreateIngress is a macro command to create a new ingress.
+func NewCmdCreateIngress(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	options := &IngressOpts{
+		CreateSubcommandOptions: NewCreateSubcommandOptions(ioStreams),
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "ingress NAME --host=hostname| --service-name=servicename [--service-port=serviceport] [--path=path] [--dry-run]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"ing"},
+		Short:                 i18n.T("Create an ingress with the specified name."),
+		Long:                  templates.LongDesc(i18n.T("Create an ingress with the specified name.")),
+		Example:               templates.Examples(i18n.T(`
+		# Create a new ingress named my-app.
+		kubectl create ingress my-app --host=foo.bar.com --service-name=my-svc`)),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	options.CreateSubcommandOptions.PrintFlags.AddFlags(cmd)
+
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, generateversioned.IngressV1GeneratorName)
+	cmd.Flags().String("host", "", i18n.T("Host name this Ingress should route traffic on"))
+	cmd.Flags().String("service-name", "", i18n.T("Service this Ingress should route traffic to"))
+	cmd.Flags().Int("service-port", 80, "Port number of the Service to route traffic to")
```

Comment:
Defaults should be set in `NewIngressOptions`.

Reply:
Each flag should be wired with appropriate member of `IngressOptions` struct.

Reply:
Fixed this accordingly

Reply:
Fixed